build_differ.py

```
@@ -21,28 +21,38 @@ def run_command(command):
+    if not arcname:
+        arcname = os.path.basename(source_dir)
+
+    print(f"Compressing {source_dir} to {target_file} with arcname {arcname}")
+
     if target_file.endswith('.tar.gz'):
         with tarfile.open(target_file, "w:gz") as tar:
-            tar.add(source_dir, arcname=os.path.basename(source_dir))
+            tar.add(source_dir, arcname=arcname)
     elif target_file.endswith('.zip'):
         with zipfile.ZipFile(target_file, 'w', zipfile.ZIP_DEFLATED) as zipf:
             for root, dirs, files in os.walk(source_dir):
```

Ruff (0.13.3)

Loop control variable `dirs` not used within loop body

Rename unused `dirs` to `_dirs`

(B007)